frontend-react/src/components/ManagePublicKey/ManagePublicKey.tsx

```diff
@@ -82,50 +99,85 @@ function ManagePublicKeySwitchDisplay() {
         }
     };
 
+    useEffect(() => {
+        if (sender && orgPublicKeys?.keys.length) {
+            // check if kid already exists for the selected org.sender
+            const kid = `${activeMembership?.parsedName}.${sender}`;
+            for (const apiKeys of orgPublicKeys.keys) {
+                if (apiKeys.keys.some((k: ApiKey) => k.kid === kid)) {
+                    setHasPublicKey(true);
+                }
+            }
+        }
+
+        if (senders?.length === 1) {
+            setSender(senders[0].name);
+            setHasBack(false);
+        }
+    }, [orgPublicKeys, sender, activeMembership?.parsedName, senders]);
+
+    const showPublicKeyConfigured =
+        sender && hasPublicKey && !uploadNewPublicKey && !fileSubmitted;
+    const showUploadMsg =
+        (sender && !fileSubmitted && !hasPublicKey) ||
+        (!uploadNewPublicKey && !sender);
+    const isUpload =
+        (sender && !fileSubmitted && !hasPublicKey) || uploadNewPublicKey;
+    const hasUploadError = fileSubmitted && !isUploading && !isSuccess;
+
     return (
-        <>
-            {/*Waiting on backend to support this
+        <GridContainer className="manage-public-key padding-bottom-5 tablet:padding-top-6">
+            {!isUploading && (
+                <h1 className="margin-top-0 margin-bottom-3">
+                    Manage public key
+                </h1>
+            )}
+            {showUploadMsg && (
+                <>
+                    <p className="font-sans-md">
+                        Send your public key to begin the REST API
+                        authentication process.
+                    </p>
+                    <SiteAlert variant="info" showIcon={false}>
+                        <Icon.Lightbulb />
+                        <span className="padding-left-1">
+                            If you need more information on generating your
+                            public key, reference page 7 in the{" "}
+                            <USLink href="/resources/programmers-guide">
+                                API Programmer’s Guide.
+                            </USLink>
+                        </span>
+                    </SiteAlert>
+                </>
+            )}
             {sender.length === 0 && (
```

Comment:
@jpandersen87 if I change from `sender.length === 0` to `sender` the logic changes since sender is an empty string. I can change to from `sender.length === 0` to `!sender`

Reply:
doh i'm dumb, that should've been `!sender && (`. sorry